src/annotator/components/ContentInfoBanner.js

```diff
@@ -8,39 +15,111 @@ import { LabeledButton, Link } from '@hypothesis/frontend-shared';
+        // Two columns in narrower viewports; three in wider
+        'grid-cols-[100px_minmax(0,auto)] gap-x-4',
+        '2xl:grid-cols-[100px_minmax(0,auto)_minmax(0,auto)] 2xl:gap-x-3'
+      )}
+    >
+      <div data-testid="content-logo">
+        {info.logo && (
+          <Link href="https://jstor.org" target="_blank">
+            <img alt={info.logo.title} src={info.logo.logo} />
           </Link>
         )}
-        {info.item.title}
-        {info.item.containerTitle && (
-          <span>
-            {' '}
-            from <i>{info.item.containerTitle}</i>
-          </span>
-        )}
-        {info.links.nextItem && (
-          <Link href={info.links.nextItem} target="_blank">
-            Next article
-          </Link>
+      </div>
+      <div
+        className={classnames(
+          // Container title (this element) is not shown on narrow screens
+          'hidden',
+          '2xl:block 2xl:whitespace-nowrap 2xl:overflow-hidden 2xl:text-ellipsis',
+          'font-semibold'
         )}
+        data-testid="content-container-info"
+        title={info.item?.containerTitle}
+      >
+        {info.item?.containerTitle}
       </div>
-      <div className="text-annotator-base">
-        <LabeledButton onClick={onClose} data-testid="close-button">
-          Close
-        </LabeledButton>
+      <div
+        className={classnames(
+          // Flex layout for item title, next and previous links
+          'flex justify-center items-center gap-x-2'
+        )}
+        data-testid="content-item-info"
+      >
+        <div
+          className={classnames(
+            // Narrower viewports center this flex content:
+            // this element is not needed for alignment
+            'hidden',
+            // Wider viewports align this flex content to the right:
+            // This empty element is needed to fill extra space at left
+            '2xl:block 2xl:grow'
+          )}
+        />
+        {info.links?.previousItem && (
+          <>
+            <Link
+              classes="flex gap-x-1 items-center text-annotator-sm whitespace-nowrap"
+              title={`Open previous item in a new window`}
+              href={info.links.previousItem}
+              underline="always"
+              target="_blank"
+              data-testid="content-previous-link"
+            >
+              <CaretLeftIcon className="w-em h-em" />
+              <span>Previous</span>
+            </Link>
+            <div className="text-annotator-sm">|</div>
+          </>
+        )}
+        <div
+          className={classnames(
+            // This element will shrink and truncate fluidly.
+            // Overriding min-width `auto` prevents the content from overflowing
+            // See https://stackoverflow.com/a/66689926/434243.
+            'min-w-0 whitespace-nowrap overflow-hidden text-ellipsis shrink font-medium'
+          )}
+        >
+          <LinkUnstyled
+            title={info.item?.title}
```

Comment:
```suggestion
            title={info.item.title}
```

`info.item` is non-nullable. The `title` property _should_ always be set. If the work is inherently untitled then some suitable placeholder will have been generated by either the LMS app or the JSTOR API.